Arrowgene.Ddon.GameServer/Handler/JobGetJobChangeListHandler.cs

```diff
@@ -19,35 +18,49 @@ public JobGetJobChangeListHandler(DdonGameServer server) : base(server)
         {
         }
 
-        public override void Handle(GameClient client, StructurePacket<C2SJobGetJobChangeListReq> packet)
+        public override S2CJobGetJobChangeListRes Handle(GameClient client, C2SJobGetJobChangeListReq request)
         {
-            S2CJobGetJobChangeListRes jobChangeList = new S2CJobGetJobChangeListRes();
-            jobChangeList.JobChangeInfo = this.buildJobChangeInfoList(client.Character);
-            jobChangeList.JobReleaseInfo = this.buildJobReleaseInfoList(client.Character);
-            jobChangeList.PawnJobChangeInfoList = client.Character.Pawns
-                .Select((pawn, index) => new CDataPawnJobChangeInfo()
-                {
-                    SlotNo = (byte) (index+1),
-                    PawnId = pawn.PawnId,
-                    JobChangeInfoList = this.buildJobChangeInfoList(pawn),
-                    JobReleaseInfoList = this.buildJobReleaseInfoList(pawn)
-                })
-                .ToList();
-            jobChangeList.PlayPointList = client.Character.PlayPointList;
-            client.Send(jobChangeList);
+            S2CJobGetJobChangeListRes res = new S2CJobGetJobChangeListRes();
+            res.JobChangeInfo = buildJobChangeInfoList(client.Character);
+            res.JobReleaseInfo = buildJobReleaseInfoList(client.Character);
+
+            if (client.Party is not null)
+            {
+                var partyPawns = client.Party.Members.Where(x =>
+                    x is PawnPartyMember pawnMember
+                    && pawnMember.Pawn.PawnType == PawnType.Main
+                    && pawnMember.Pawn.CharacterId == client.Character.CharacterId)
+                    .Select(x => ((PawnPartyMember)x).Pawn)
+                    .ToList();
+
+                res.PawnJobChangeInfoList = client.Character.Pawns
+                    .Where(x => partyPawns.Contains(x))
+                    .Select((pawn, index) => new CDataPawnJobChangeInfo()
+                    {
+                        SlotNo = (byte)(index + 1),
```

Comment:
The index is done after the filtering, so if you got 3 pawns, A, B and C and only two of them, A and C in the party, they would be reported as slots 1 and 2 when they should be slots 1 and 3